std/algorithm/comparison.d

```diff
@@ -1565,3 +1568,228 @@ unittest
                 2, "two",
                 ));
 }
+
+/**
+Returns true if all of the elements in $(D r1) are present in $(D r2). Otherwise,
+returns false.
+
+If your parameters are forward ranges, then this function can allocate if the
+$(D AllocateGC.yes) flag is passed. This has the benefit of being much
+faster than the $(D AllocateGC.no) option.
```

Comment:
Define "much faster" in terms of complexity:

With AllocateGC.yes, complexity is amortized O(n)
With AllocateGC.no, complexity is O(n) for identical permutations, and up to O(n^2) for different permutations.